hrabsence/CRM/HRAbsence/Form/AbsenceRequest.php

```diff
@@ -517,320 +541,410 @@ static function formRule($fields, $files, $self) {
-          'source_record_id' => $submitValues['source_record_id'],
-          'option.limit' => 365,
-        ));
-        foreach ($result['values'] as $row_result) {
-          civicrm_api3('Activity', 'delete', array(
-            'id' => $row_result['id'],
-          ));
-        }
-        foreach ($absentDateDurations as $date => $duration) {
-          $result = civicrm_api3('Activity', 'create', array(
-            'activity_type_id' => CRM_Core_OptionGroup::getValue('activity_type', 'Absence', 'name'),
-            'source_record_id' => $submitValues['source_record_id'],
-            'activity_date_time' => $date,
-            'duration' => $duration['duration'],
-            'status_id' => $duration['approval'],
-          ));
-        }
+      } else {
         $buttonName = $this->controller->getButtonName();
         if ($buttonName == "_qf_AbsenceRequest_done_save") {
-          $this->_aid = $submitValues['source_record_id'];
-          $sendTemplateParams['from'] = "{$mailprm[$this->_targetContactID]['display_name']} <$domainEmailAddress>";
+          $sendTemplateParams['from'] = $targetContactEmail;
           $sendTemplateParams['tplParams']['save'] = $sendMail = TRUE;
-          $session->pushUserContext(CRM_Utils_System::url('civicrm/absences', "reset=1&cid={$this->_targetContactID}", false, 'hrabsence/list'));
         }
       }
-      if (!empty($submitValues['hidden_custom'])) {
-        $customFields = CRM_Utils_Array::crmArrayMerge(
-          CRM_Core_BAO_CustomField::getFields('Activity', FALSE, FALSE, $this->_activityTypeID),
-          CRM_Core_BAO_CustomField::getFields('Activity', FALSE, FALSE, NULL, NULL, TRUE)
-        );
-        $customValues = CRM_Core_BAO_CustomField::postProcess($submitValues, $customFields, $result['id'], 'Activity');
-        CRM_Core_BAO_CustomValueTable::store($customValues, 'civicrm_activity', $result['id']);
-      }
+    }
 
-      if (!empty($customValues)) {
-        $customGroup = array();
-        foreach ($customValues as $fieldID => $values) {
-          foreach ($values as $fieldValue) {
-            $customValue = array('data' => $fieldValue['value']);
-            $customFields[$fieldID]['id'] = $fieldID;
-            $formattedValue = CRM_Core_BAO_CustomGroup::formatCustomValues($customValue, $customFields[$fieldID], TRUE);
-            $customGroup[$customFields[$fieldID]['groupTitle']][$customFields[$fieldID]['label']] = str_replace('&nbsp;', '', $formattedValue);
-          }
+    if (empty($sendTemplateParams['from'])) {
+      $sendTemplateParams['from'] = "$domainEmailAddress";
+    }
+
+    $sendTemplateParams['sendMail'] = $sendMail;
+
+    return $sendTemplateParams;
+  }
+
+  /**
+   * Creates main activity for absence request and subactivities for each absence
+   * day in given activity status.
+   * 
+   * @param array $absentDateDurations
+   *   List of durations (days) of leave
+   * @param type $status
```

Comment:
I guess the $status paramter should string ... you are using (type) which is undefined type